quint/src/effects/builtinSignatures.ts

```diff
@@ -223,6 +223,7 @@ const otherOperators = [
   { name: 'reps', effect: parseAndQuantify('(Pure, (Read[r1]) => Read[r2] & Update[u]) => Read[r1, r2] & Update[u]') },
   { name: 'fail', effect: propagateComponents(['read', 'update'])(1) },
   { name: 'assert', effect: propagateComponents(['read'])(1) },
+  { name: 'debug', effect: propagateComponents(['read'])(2) },
```

Comment:
If `debug` is indeed pure, do we need to specify that here?

Reply:
`propagateComponents(['read'])` is compatible with a pure def, and I believe this is the most appropriate effect. We probably don't want to call `debug` on actions (i.e. `debug("foo", x' = x + 1)`) nor in temporal formulas (i.e. `debug("bar", next(x) == x)`)